ide/che-core-ide-app/src/main/java/org/eclipse/che/ide/resources/impl/ResourceManager.java

```diff
@@ -1171,45 +1170,38 @@ private Resource newResourceFrom(final ItemReference reference) {
   }
 
   protected Promise<ResourceDelta[]> synchronize(final ResourceDelta[] deltas) {
-    List<Promise<Void>> promisesToResolve = new ArrayList<>(deltas.length);
+    Promise<Void> chain = promises.resolve(null);
```

Comment:
I am not follow the logic here. 
`Promise<Void> chain = promises.resolve(null); `
then `chain = chain.thenPromise(something)`;
What you want to solve here?

Reply:
Okay, I'll explain. If we'll have the code like this:

```